Added IO, copy and show tests.

vinctux opened the pull request:
Summary:

- Added IOStream test (via a pseudo-temporary file).
- Added `copy` test (just a `try`).
- Added `show` test (simple REPL string comparison).
- Fixed deprecated `readbytes(f)` to `read(f, UInt8, 1)`, which should be identical.

tkelman commented on test/runtests.jl:
```diff
@@ -207,8 +207,12 @@ for sha_idx in 1:length(sha_funcs)
 end
 println("Done! [$(nerrors - nerrors_old) errors]")
 
+if VERSION > v"0.5.0-dev"
```
be more specific about version cutoffs

staticfloat commented on test/runtests.jl:
```diff
@@ -181,6 +207,38 @@ for sha_idx in 1:length(sha_funcs)
+        copy(ctxs[idx]())
+    catch
+        print("\n")
+        warn("Some weird copy error happend with $(ctxs[idx])")
```
`happend` -> `happened`

staticfloat commented on test/runtests.jl:
```diff
@@ -181,6 +207,38 @@ for sha_idx in 1:length(sha_funcs)
+    print("Testing show function @ $(ctxs[idx]) ...")
+    if replstr(ctxs[idx]()) != shws[idx]
+        print("\n")
+        warn("Some weird show error happend with $(ctxs[idx])")
```
`happend` -> `happened`

staticfloat commented:
I merged another PR before this one, can you please rebase and address the minor comments in the diff above?

vinctux commented:
I'm really sorry but I just can't make it work... I rebased my 3 commits into one, fixed the typo, committed the change, rebased again, pushed, but it still seems that it can't be automatically merged. What should I do?

staticfloat commented:
Try rebasing this branch on top of the current state of `origin/master`.  You can do that with `git pull --rebase origin master`.  This will undo your changes, pull in the commits from `origin/master`, then try to replay your diffs on top of that new base.  This may have some kind of merge conflict which you will have to fix by manually editing the files that are conflicting, then you finish the rebase with `git add <conflicted file path>` and `git commit`.  Whenever there is a git merge conflict, you can always run `git status` to see what the status of the files are, and what commands you need to run to mark something as manually merged.

After that is all finished, you push up your changes to this remote branch that I can merge in, but since you will have changed the history of the branch, you will need to force-push, (e.g. add `-f` to the end of your `git push`).

vinctux commented:
@staticfloat Thank you very much, I now do understand how it works. But as it seems I messed something so severely up, that I can't use your instructions. I hope you don't mind that I close this PR and open another PR (identical changes) in #21. Sorry for the mess.

staticfloat commented:
Yep, no worries.  `git` takes a while to get fluent with.
